### PR TITLE
feat: label streaming swap txs

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

Add a "Streaming Swap" label to swap TXs that are streaming (detected via the memo).

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

Closes https://github.com/shapeshift/web/issues/5092

## Risk

Small

## Testing

Streaming swaps should now be labelled as "Streaming Swap" instead of "Swap" in transaction history.

### Engineering

☝️

### Operations

☝️

## Screenshots (if applicable)

Screenshot 2023-09-04 at 10 28 02 am

### CODE REVIEW

**Comment by 0xApotheosis** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5209** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5209" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5209?utm_source=stack-comment).

File: src/components/TransactionHistoryRows/TransactionMethod.tsx [hunk 1]
```diff
@@ -18,6 +18,11 @@ import { TransactionGenericRow } from './TransactionGenericRow'
 import type { TransactionRowProps } from './TransactionRow'
 import { getTransfersByType, getTxMetadataWithAssetId } from './utils'
 
+const isStreamingSwapMemo = (memo: string): boolean => {
+  const regex = /:\d+\/\d+\/\d+:/
+  return regex.test(memo)
+}
```
Comment (gomesalexandre) [inline src/components/TransactionHistoryRows/TransactionMethod.tsx:24]: LGTM 🤌 
image

File: src/components/TransactionHistoryRows/TransactionMethod.tsx [hunk 2]
```diff
@@ -28,6 +33,9 @@ export const TransactionMethod = ({
 }: TransactionRowProps) => {
   const translate = useTranslate()
   const txMetadata = useMemo(() => txDetails.tx.data!, [txDetails.tx.data]) // we are guaranteed to have had metadata to render this component
+  const memo = (txDetails.tx.data as { memo: string }).memo // TxMetadata typing is broken
```
Comment (gomesalexandre) [inline src/components/TransactionHistoryRows/TransactionMethod.tsx:36]: Should be happy as long as we properly discriminate cosmos `SwapMetadata` i.e:

https://github.com/shapeshift/web/blob/93b4fabdffed4ef29d9b34ff37348462d92dfca2/packages/unchained-client/src/cosmossdk/parser/types.ts#L24-L27

image

Comment (gomesalexandre) [reply src/components/TransactionHistoryRows/TransactionMethod.tsx]: 2309ee5ff6

**Review by gomesalexandre — APPROVED** [review]
Tested locally, does what it says on the box:

- Streaming swap

image
image

- Regular good ol' THOR swap

image
image

File: src/components/TransactionHistoryRows/TransactionMethod.tsx (hunk 2 above)
Comment (0xApotheosis) [reply src/components/TransactionHistoryRows/TransactionMethod.tsx]: A much bigger 🧠 approach, gm ser

**Pull request merged**